app_feup/lib/view/Widgets/schedule_slot.dart

```diff
@@ -95,8 +95,14 @@ class ScheduleSlot extends StatelessWidget {
   }
 
   Widget createScheduleSlotTeacherInfo(context) {
+    final int maxSizeTeacher = 14;
+    String teacher = this.teacher.replaceAll(
+        RegExp('[A-Z]*-',
+            unicode: true),'');
+    teacher = teacher.length > maxSizeTeacher ? teacher.substring(0,10) + '...'
+        : teacher;
     return createTextField(
-        this.teacher,
+        teacher,
```

Comment:
The overflow API can be used to make this instead of manually. Can you make that change?

https://api.flutter.dev/flutter/painting/TextOverflow.html